### ISSUE TITLE
virtualbox-ovf builder can't attach Guest Additions to SATA CD drive

### ISSUE THREAD

**Issue by nsballmann** [hashicorp/packer#7249]
I'm using 

    packer@heppo:~/jw/workspace$ packer --version
    1.3.3

And I'm trying to use the `virtualbox-ovf` builder. I'm trying to use `"guest_additions_mode": "attach",`. Validation says I'm fine:

    + packer validate -var version=trunk ./<some>.json
    Template validated successfully.

But when I'm building I get the following:

    + packer build -timestamp-ui -color=false -force -var version=trunk ./<some>.json
    ==> virtualbox-ovf: Retrieving Guest additions
        virtualbox-ovf: Using file in-place: file:///usr/share/virtualbox/VBoxGuestAdditions.iso
    ==> virtualbox-ovf: Retrieving OVF/OVA
        virtualbox-ovf: Using file in-place: file:///home/packer/jw/workspace/<build>/src/<some>.ova
    ==> virtualbox-ovf: Importing VM: /home/packer/jw/workspace/<build>/src/<some>.ova
    ==> virtualbox-ovf: Error attaching guest additions: VBoxManage error: VBoxManage: error: Could not find a controller named 'IDE Controller'
    ==> virtualbox-ovf: Deregistering and deleting imported VM...
    ==> virtualbox-ovf: Deleting output directory...
    Build 'virtualbox-ovf' errored: Error attaching guest additions: VBoxManage error: VBoxManage: error: Could not find a controller named 'IDE Controller'
    
    ==> Some builds didn't complete successfully and had errors:
    --> virtualbox-ovf: Error attaching guest additions: VBoxManage error: VBoxManage: error: Could not find a controller named 'IDE Controller'
    
    ==> Builds finished but no artifacts were created.

So the HD and CD drives in the imported ova file are attached to a SATA controller. I saw the `virtualbox-iso` builder has `"iso_interface": "sata"` and `"hard_drive_interface": "sata"`. I understand those are for newly generated VMs. I thought maybe they work in the `virtualbox-ovf` builder too, but no luck:
    
    + packer validate -var version=trunk ./<some>.json
    Template validation failed. Errors are shown below.
    
    Errors validating build 'virtualbox-ovf'. 2 error(s) occurred:
    
    * unknown configuration key: "hard_drive_interface"
    * unknown configuration key: "iso_interface"

Knowing those were not listed [in the documentation](https://www.packer.io/docs/builders/virtualbox-ovf.html), I kind of suspected this result. But now I'm stuck.

How do I specify the CD drive interface for the guest additions in the `virtualbox-ovf` builder?

I don't think the host platform is necessary for this case, but anyway:

    packer@heppo:~/jw/workspace$ uname -a
    Linux heppo 4.15.0-43-generic #46~16.04.1-Ubuntu SMP Fri Dec 7 13:31:08 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
    packer@heppo:~/jw/workspace$ lsb_release -a
    No LSB modules are available.
    Distributor ID: Ubuntu
    Description:    Ubuntu 16.04.5 LTS
    Release:        16.04
    Codename:       xenial

If needed i'll try to provide a minimal working example.

**Comment by rgl** [comment]
FYI, the virtualbox-iso does not honour iso_interface, see https://github.com/hashicorp/packer/issues/7256

**Comment by nsballmann** [comment]
I helped myself like this: 

```json
"vboxmanage": [
    ["storageattach", "{{.Name}}", "--storagectl", "SATA", "--port", "2", "--type", "dvddrive", "--medium", "additions"]
],
"vboxmanage_post": [
    ["storageattach", "{{.Name}}", "--storagectl", "SATA", "--port", "2", "--type", "dvddrive", "--medium", "emptydrive"],
 ]
```

But this is just a workaround...

**Comment by SwampDragons** [comment]
This should be fixed by #7298 and there are some binaries available on that PR -- can you test it out and lmk if it works for you?

**Comment by rgl** [comment]
@SwampDragons it works for me! thanks :-)
